components/token-container.tsx

```diff
@@ -56,9 +92,18 @@ export const TokenContainer = (props: TokenContainerProps) => {
               display={"flex"}
               h={"fit-content"}
               p={"2px 6px"}
-              onClick={() => {
-                if (props.maxBalance) {
-                  props.setAmount(props.maxBalance);
+              onClick={async () => {
```

Comment:
dont pass entire functions like that to the onClick prop, create a handler function above the return statement that renders jsx and simply reference it in onClick. This is easier to read and much better in terms of performance

Reply:
Done